src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,43 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents a task with a deadline
+ */
+public class Deadline extends Task {
+
+    protected LocalDate date;
+
+    /**
+     * Deadline constructor.
+     *
+     * @param description Description of deadline task.
+     * @param date Date of deadline.
+     */
+    public Deadline(String description, LocalDate date) {
+        super(description);
+        this.date = date;
+    }
+
+    /**
+     * Returns a Deadline from string input with description and date.
+     *
+     * @param desc_date String input with description and date in
+     *                  '[description] (by: [MMM d yyyy]' format.
+     * @return Deadline from string input with description and date.
+     */
+    public static Deadline build(String desc_date) {
```

Comment:
I think you should change this to be camel case.
```suggestion
    public static Deadline build(String descriptionAndDate) {
```